### PR TITLE
pkp/pkp-lib#2407 Remove mutex work-around for UNION queries

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

**Comment by asmecher** [comment]
@bozana, just a reminder about this one!

File: classes/oai/PKPOAIDAO.inc.php [hunk 1]
```diff
@@ -37,43 +37,69 @@ function setOAI($oai) {
...
+		);
+	}
 
 	/**
-	 * Return the *nix timestamp of the earliest published submission.
-	 * @param $setIds array optional Objects ids that specify an OAI set,
-	 * in hierarchical order. If empty, all records from
-	 * all sets will be included.
-	 * @return int
+	 * Retrieve a resumption token.
```
Comment (bozana) [inline classes/oai/PKPOAIDAO.inc.php]: @param is missing

File: classes/oai/PKPOAIDAO.inc.php [hunk 2]
```diff
@@ -107,21 +120,11 @@ function recordExists($dataObjectId, $setIds = array()) {
...
-		);
-
-		$returner = null;
+		$result = $this->_getRecordsRecordSet($setIds, null, null, null, $dataObjectId);
 		if ($result->RecordCount() != 0) {
 			$row = $result->GetRowAssoc(false);
 			$returner = $this->_returnRecordFromRow($row);
-		}
-
+		} else $record = null;
```
Comment (bozana) [inline classes/oai/PKPOAIDAO.inc.php]: $record or $returner = null ?

File: classes/oai/PKPOAIDAO.inc.php [hunk 3]
```diff
@@ -171,192 +171,37 @@ function getRecords($setIds, $from, $until, $set, $offset, $limit, &$total) {
...
-		// Add the set specification, if any.
-		if (isset($set)) {
-			$params[] = $set;
-		}
-
-		// Add the set ids again, so they can be used in the tombstone JOIN part of the sql too.
-		$params = array_merge($params, $notNullSetIds);
-
-		return $params;
+		return OAIUtils::UTCtoTimestamp($returner->datestamp, false);
```
Comment (bozana) [inline classes/oai/PKPOAIDAO.inc.php]: if $returner is null, is it ok to access the datestamp like this? :-\

File: classes/oai/PKPOAIDAO.inc.php (hunk 2 above)
Comment (asmecher) [inline classes/oai/PKPOAIDAO.inc.php]: Oops, fixed!

File: classes/oai/PKPOAIDAO.inc.php (hunk 1 above)
Comment (asmecher) [inline classes/oai/PKPOAIDAO.inc.php]: Added, thanks!

File: classes/oai/PKPOAIDAO.inc.php (hunk 3 above)
Comment (asmecher) [inline classes/oai/PKPOAIDAO.inc.php]: Nope -- fixed, thanks!

**Pull request merged**